p2p/transport/websocket/websocket.go

```diff
@@ -188,7 +200,14 @@ func (t *WebsocketTransport) maDial(ctx context.Context, raddr ma.Multiaddr) (ma
 
 	transport := &http.Transport{
 		DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
-			conn, err := net.Dial(network, addr)
+			var conn manet.Conn
+			var err error
+			if t.UseReuseport() {
+				conn, err = t.reuse.Dial(raddr)
+			} else {
+				var d manet.Dialer
+				conn, err = d.Dial(raddr)
```

Comment:
Same here? By the way, what's the advantage of replacing `net.Dial` here? Not saying this is wrong, just curious.

Reply:
Sure, will change to DialContext. 
No functional advantage. Since there is already a wrapper for the dialer that uses multiaddr, wanted to use the same. Only minor advantage of use this wrapped dialer is in future if dialer is enhanced to add more capabilities, usage need not be changed here to take advatange of it.